### ISSUE TITLE
Re-import detection should permit partial and multiple overlap

### ISSUE THREAD

**Issue by sampsyo** [beetbox/beets#702]
_Re-imports_ occur when the user wants to re-tag some files that are already in their library. Currently, an album is considered a re-import only when it has exactly the same set of paths as an existing album. We should also permit re-imports that:
- Partially overlap with an existing album. This happens when the user wants to add tracks to an existing album, as in #112.
- Consist of multiple albums. This happens when the user erroneously imported a multi-disc album as separate albums initially but wants to re-import them together (e.g., with `--flat`).

The point where this is detected is in `beets.importer._duplicate_check`. We also need to support the appropriate replacement action in `apply_choices`.

This is especially pernicious since the user sees a "this album is already in the library!" prompt when the re-import check fails. If they choose "remove old", then some of the files being imported are deleted. No good!

**Comment by discopatrick** [comment]
+1 - I would love this to be fixed.

I have an issue in the same sort of area; I wonder if it should be added to this issue or a separate one opened?:

I am currently redownloading my entire eMusic collection since they reuploaded every single release as 320k mp3 (compared to VBR prior to this). When I re-import these albums, and the autotagger shows me the list of possible matches, I want beets to flag to me which of the suggestions are already in my collection - perhaps with a green dot or something. This is especially important when a release has multiple versions that are very similar - without this flag, I won't know which release I chose the first time I imported.

**Comment by sampsyo** [comment]
Yes, that's a very reasonable suggestion—in fact, I think that's the idea behind https://github.com/beetbox/beets/issues/351.
